TundraTD/Assets/Scripts/Level/MobPool.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Mobs;
+using Mobs.MobsBehaviour;
+using Mobs.MobsBehaviour.Bear;
+using Mobs.MobsBehaviour.Boar;
+using Mobs.MobsBehaviour.Spider;
+using UnityEngine;
+
+namespace Level
+{
+    public class MobPool : MonoBehaviour
+    {
+        private List<MobBehaviour> _mobsOnLevel;
+
+        private static int _boarStartIndexMarker;
+        private static int _bearStartIndexMarker;
+        private static int _spiderStartIndexMarker;
+
+        private void Start()
+        {
+            _mobsOnLevel = new List<MobBehaviour>();
+        }
+
+        public void InstantiateMob(MobBehaviour mob) => _mobsOnLevel.Add(mob);
+
+        public bool AreAllMobDead()
+        {
+            return _mobsOnLevel.All(mobs => !mobs.MobModel.IsAlive);
+        }
+
+        public void EndMobInstantiation()
+        {
+            _mobsOnLevel.Sort((x, y) => OrderOnType(x).CompareTo(OrderOnType(y)));
+            _boarStartIndexMarker = _mobsOnLevel.FindIndex(x => x is BoarBehaviour);
+            _bearStartIndexMarker = _mobsOnLevel.FindIndex(x => x is BearBehaviour);
+            _spiderStartIndexMarker = _mobsOnLevel.FindIndex(x => x is SpiderBehaviour);
+            
+            int OrderOnType(MobBehaviour item)
+            {
+                switch (item)
+                {
+                    case BoarBehaviour _:
+                        return 0;
+                    case BearBehaviour _:
+                        return 1;
+                    case SpiderBehaviour _:
+                        return 2;
+                    default:
+                        return -1;
```

Comment:
Well, this thing looks a bit strange too. I don't understand "_" and this logic completely ¯\_(ツ)_/¯

Reply:
Function OrderOnType was created to sort mobs based on their class type. We can create an ID for each mob but it will change line 34

_mobsOnLevel.Sort((x, y) => OrderOnType(x).CompareTo(OrderOnType(y)));
_mobsOnLevel.Sort((x, y) => OrderOnType(x).CompareTo(x => x.id));